src/main/drivers/bus_spi.c

```diff
@@ -17,89 +17,48 @@
 
 #include <stdbool.h>
 #include <stdint.h>
+#include <string.h>
 
 #include <platform.h>
 
+#include "build/debug.h"
+
 #include "drivers/bus_spi.h"
+#include "drivers/bus_spi_impl.h"
```

Comment:
I don't see the advantage of having a separate `bus_spi_impl.h` - I would just have `bus_spi.h`. This makes the code and the PR easier to read.